File: matlive-sdk/src/main/java/com/matnsolutions/matlive_sdk/audio/seats/mat_live_audio_room_layout_config.dart
class MatLiveAudioRoomLayoutConfig {
  MatLiveAudioRoomLayoutConfig({
    this.rowSpacing = 0,
    List<MatLiveAudioRoomLayoutRowConfig>? rowConfigs,
  }) : rowConfigs = rowConfigs ??
            [
              MatLiveAudioRoomLayoutRowConfig(),
              MatLiveAudioRoomLayoutRowConfig(),
            ];

  /// The spacing between rows, in pixels
  final double rowSpacing;

  /// Configuration of each row
  List<MatLiveAudioRoomLayoutRowConfig> rowConfigs;

  @override
  String toString() {
    return 'rowSpacing:$rowSpacing, rowConfigs:${rowConfigs.toString()}';
  }
}

/// Configuration of the row in the seat layout
class MatLiveAudioRoomLayoutRowConfig {
  MatLiveAudioRoomLayoutRowConfig({
    this.count = 5,
    this.seatSpacing = 0,
  });
  int count;

  int seatSpacing = 0;

  @override
  String toString() {
    return 'row config:{count:$count, spacing:$seatSpacing}';
  }
}

File: matlive-sdk/src/main/java/com/matnsolutions/matlive_sdk/audio/seats/MatLiveAudioRoomLayoutConfig.kt
package com.matnsolutions.matlive_sdk.audio.seats

data class MatLiveAudioRoomLayoutConfig(
    val rowSpacing: Double = 0.0,
    val rowConfigs: List<MatLiveAudioRoomLayoutRowConfig> = listOf(
        MatLiveAudioRoomLayoutRowConfig(),
        MatLiveAudioRoomLayoutRowConfig()
    )
) {
    override fun toString(): String {
        return "rowSpacing:$rowSpacing, rowConfigs:${rowConfigs.toString()}"
    }
}
